libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -1198,6 +1198,38 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
         assertFalse(mapboxNavigation.getRerouteController() === oldRerouteController)
     }
 
+    @Test
+    fun `set null reroute controller`() = coroutineRule.runBlockingTest {
+        val oldController: RerouteController = mockk(relaxed = true)
+        val newController: RerouteController? = null
+        createMapboxNavigation()
+        mapboxNavigation.setRerouteController(oldController)
+
+        mapboxNavigation.setRerouteController(newController)
+
+        assertNull(mapboxNavigation.getRerouteController())
+    }
+
+    @Test
+    fun `set navigation reroute controller`() = coroutineRule.runBlockingTest {
+        val navigationRerouteController: NavigationRerouteController = mockk(relaxed = true)
+        createMapboxNavigation()
+        mapboxNavigation.setRerouteController(navigationRerouteController)
+        assertEquals(navigationRerouteController, mapboxNavigation.getRerouteController())
+    }
+
+    @Test
+    fun `set null navigation reroute controller`() = coroutineRule.runBlockingTest {
+        val oldController: NavigationRerouteController = mockk(relaxed = true)
+        val newController: NavigationRerouteController? = null
+        createMapboxNavigation()
+        mapboxNavigation.setRerouteController(oldController)
+
+        mapboxNavigation.setRerouteController(newController)
```

Comment:
nit: actions together
```suggestion

        mapboxNavigation.setRerouteController(oldController)
        mapboxNavigation.setRerouteController(newController)
```

Reply:
It's intentional. First `setRerouteController` invocation is more about preparing. I want to check what happens when I set another one. I's vague but I see it this way. Maybe I'm wrong. What do you usually do in these situations?

Reply:
good point actually to a discussion. don't see any against your view. We can keep it as initially set 👍